File: Java/Classes/Entity.java
public abstract class Entity {
    private String name;
    private String description;
    private int id;

    // Constructor
    public Entity(String name, String description, int id) {
        this.name = name;
        this.description = description;
        this.id = id;
    }

    // Getters
    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public int getId() {
        return id;
    }

    // Returns basic entity information
    public String getEntityInfo() {
        return "ID: " + id + ", Name: " + name + ", Description: " + description;
    }

    // Abstract method to be implemented by subclasses
    public abstract String getDetails();

    // Override toString to provide all information
    @Override
    public String toString() {
        return getEntityInfo() + "\n" + getDetails();
    }
}

File: C++/cpp_files/Entity.cpp
#include "Entity.h"

Entity::Entity(std::string name, std::string description, int id)
    : name(std::move(name)), description(std::move(description)), id(id) {
}

Entity::~Entity() = default;

std::string Entity::getName() const {
    return name;
}

std::string Entity::getDescription() const {
    return description;
}

int Entity::getId() const {
    return id;
}

std::string Entity::getEntityInfo() const {
    return "ID: " + std::to_string(id) + "\nName: " + name + "\nDescription: " + description;
}

std::string Entity::toString() const {
    return getEntityInfo() + "\n" + getDetails();
}